### ISSUE TITLE
[Task]: fix unexpected regex differences between old unEvalTree and newUnEvalTree

### ISSUE THREAD

**Issue by Rishabh-Rathod** [appsmithorg/appsmith#12582]
### Is there an existing issue for this?

- [X] I have searched the existing issues

Issue reported by @SatishGandham on [slack](https://theappsmith.slack.com/archives/C02K0SZQ7V3/p1648809100362959)

<img width="868" alt="Screenshot 2022-04-05 at 1 27 31 AM" src="https://user-images.githubusercontent.com/23132741/161704037-96601240-4ca5-4851-904b-a2088ced7166.png">

**Comment by somangshu** [comment]
@Rishabh-Rathod please add an estimate here

**Comment by Rishabh-Rathod** [comment]
> @Rishabh-Rathod please add an estimate here
> Also, please add any profiling benchmarks to the description. It's our argument as to why we are picking this up

Hey @somangshu, as the core issue for this change was a bug https://github.com/appsmithorg/appsmith/issues/12520  where I added estimate for this and the above-mentioned issue was a side-effect of it which @SatishGandham noticed. 

I opened this issue to keep track of that. It is not a new performance task but maybe a regression after this https://github.com/appsmithorg/appsmith/pull/11001.

I have added a few details in the PR related to the benchmark of the library we will use now, I will add more details in the same PR.
